scripts/client.py

```diff
@@ -137,15 +170,56 @@ def list_species(self, organization_id):
     def create_seedling_batch(self, payload):
         return self.post("/api/v1/nursery/batches", json=payload)["batch"]
 
+    def fetch_access_token(self):
+        if self.refresh_token:
+            # This depends on how Keycloak populates some JWT fields. We don't bother verifying
+            # the signature because we're only using this to form a request to send to Keycloak,
+            # and Keycloak will reject it if the parameters are bogus.
+            decoded_token = jwt.decode(
+                self.refresh_token, options={"verify_signature": False}
+            )
+            base_url = decoded_token["iss"]
+
+            r = requests.post(
+                f"{base_url}/protocol/openid-connect/token",
+                data={
+                    "client_id": self.client_id,
```

Comment:
just in case it is possible.. can client id be inferred from `sub` in decoded jwt? i can't recall if that only shows up in the access token.
this is fine too.. but just in case it is possible.. and we want client_id to be optional in runtime args - when refresh token is provided.

Reply:
Ah, good point. It can indeed.

Reply:
For the record, Keycloak puts it in the `azp` (audience) field.